src/table/use-sticky-columns.ts

```diff
@@ -0,0 +1,362 @@
+  useEffect(
+    () => {
+      if (!stickyColumns.isEnabled) {
+        return;
+      }
+
+      const selector = (state: StickyColumnsState) => state.cellState[columnId];
+
+      const updateCellStyles = (state: null | StickyColumnsCellState) => {
+        const className = getClassName(state);
+        const cellElement = cellRef.current;
+        if (cellElement) {
+          Object.keys(className).forEach(key => {
+            if (className[key]) {
+              cellElement.classList.add(key);
+            } else {
+              cellElement.classList.remove(key);
+            }
+          });
+          cellElement.style.left = state?.offset?.left !== undefined ? `${state?.offset?.left}px` : '';
+          cellElement.style.right = state?.offset?.right !== undefined ? `${state?.offset?.right}px` : '';
+        }
+      };
+
+      const unsubscribe = stickyColumns.store.subscribe(selector, newState => updateCellStyles(selector(newState)));
+      return unsubscribe;
+    },
+    // getClassName is expected to be pure
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [stickyColumns.store, stickyColumns.isEnabled, columnId]
+  );
+
+  // Provide cell styles as props so that a re-render won't cause invalidation.
+  const cellStyles = stickyColumns.store.get().cellState[columnId];
+  return {
+    ref: refCallback,
+    className: cellStyles ? clsx(getClassName(cellStyles)) : undefined,
+    style: cellStyles?.offset ?? undefined,
+  };
+}
+
+interface UpdateCellStylesProps {
+  wrapper: HTMLElement;
+  table: HTMLElement;
+  cells: Record<ColumnId, HTMLElement>;
+  visibleColumns: readonly ColumnId[];
+  stickyColumnsFirst: number;
+  stickyColumnsLast: number;
+}
+
+export default class StickyColumnsStore extends AsyncStore<StickyColumnsState> {
+  private cellOffsets = new Map<ColumnId, { first: number; last: number }>();
+  private stickyWidthLeft = 0;
+  private stickyWidthRight = 0;
+  private isStuckToTheLeft = false;
+  private isStuckToTheRight = false;
+
+  constructor() {
+    super({ cellState: {}, scrollPadding: { left: 0, right: 0 } });
+  }
+
+  public updateCellStyles(props: UpdateCellStylesProps) {
+    const hasStickyColumns = props.stickyColumnsFirst + props.stickyColumnsLast > 0;
+    const hadStickyColumns = this.cellOffsets.size > 0;
+
+    if (hasStickyColumns || hadStickyColumns) {
+      this.updateScroll(props);
+      this.updateCellOffsets(props);
+      this.set(() => ({
+        cellState: this.generateCellStyles(props),
+        scrollPadding: { left: this.stickyWidthLeft, right: this.stickyWidthRight },
+      }));
+    }
+  }
+
+  private updateScroll(props: UpdateCellStylesProps) {
+    const wrapperScrollLeft = props.wrapper.scrollLeft;
+    const wrapperScrollWidth = props.wrapper.scrollWidth;
+    const wrapperClientWidth = props.wrapper.clientWidth;
+    const tablePaddingLeft = parseInt(getComputedStyle(props.table).paddingLeft) || 0;
```

Comment:
I'd suggest to use `parseFloat` here instead of `parseInt`, to keep the precision of the value.

(In case you want to keep `parseInt`, I recommend adding the `radix` parameter to it to prevent any accidental conversion, see [MDN docs on parseInt](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt))